### ISSUE TITLE
get rid of LocaleCalendar

### ISSUE THREAD

**Issue by imposeren** [wreckage/django-happenings#15]
LocaleHTMLCalendar uses context manager with this code:

```
class TimeEncoding:
    def __init__(self, locale):
        self.locale = locale

    def __enter__(self):
        self.oldlocale = _locale.getlocale(_locale.LC_TIME)
        _locale.setlocale(_locale.LC_TIME, self.locale)
        return _locale.getlocale(_locale.LC_TIME)[1]

    def __exit__(self, *args):
        _locale.setlocale(_locale.LC_TIME, self.oldlocale)

```

this is not thread safe: if 2 calendars using different locale settings are rendered during the same time then one of them may have wrong locale used because of how ``_locale.setlocale`` works.

This is not a problem now because locale of calendar can't be changed. But if someone is willing to implement multi-language site and will subclass EventCalendar to do so - he will encounter a lot of unpredictable problems when users with different language will access site at the same time.

Possible solutions:
* use https://pypi.python.org/pypi/PyICU/ for thread-safe localization. But it have dependency on C++ library
* use django i18n features and provide translations for all locales. Django seems already have translations to all that we need so maybe no additional translation will be required: https://github.com/django/django/blob/master/django/utils/dates.py , https://github.com/django/django/blob/master/django/conf/locale/ru/LC_MESSAGES/django.po#L816
* just do not use ``calendar.month_name[themonth]``: use django template filter ``|date`` with appropriate formats because they are already using localization from previous suggested solution. Or use ``django.utils.dates.MONTHS[month_num]``.Other uses of localization from ``calendar`` module (if any)  should also be removed. ``locale.setlocale`` in ``utils.displays`` should also be removed

What do you think?

**Comment by wreckage** [comment]
I'd like to avoid any unnecessary dependencies (which is why the app was created with so few dependencies), so that leaves out the first solution... Both 2nd and 3rd solutions seem like they'd work great... Whichever of the two that is easiest to implement would probably be the way to go. What do you feel is the best solution?

**Comment by imposeren** [comment]
I think that easiest solution will be to pass ``date`` or ``datetime`` instance to template and use ``{{ current_date|date:"F" }}`` in template instead of ``{{ display_month }}``.

That way other projects will have full control over what is displayed and looks like it will be almos 100% backward compatible

**Comment by imposeren** [comment]
this is implemented in merged #16 and this issue should be closed
